File: Climbing_Stairs/climb.dart
import 'dart:collection';

class Solution {
  HashMap<int, int> memoir = HashMap<int, int>();
  int climbStairs(int n) {
    if (memoir.containsKey(n)) {
      return memoir[n]!;
    }

    if (n <= 2) {
      return n;
    }

    memoir[n] = climbStairs(n - 1) + climbStairs(n - 2);

    return memoir[n]!;
  }
}

File: Climbing_Stairs/climb.go
package climbingstairs

var memoir = make(map[int]int)

func climbStairs(n int) int {
	if val, exist := memoir[n]; exist {
		return val
	}

	if n <= 2 {
		return n
	}

	memoir[n] = climbStairs(n-1) + climbStairs(n-2)

	return memoir[n]
}
